test/screenshot/lib/controller.js

```diff
@@ -252,17 +252,109 @@ class Controller {
       .then(
         (image) => {
           return new Promise((resolve, reject) => {
+            const {rows, cols} = this.getCropMatches_(image);
+            const cropRect = this.getCropRect_({rows, cols});
+            const {x, y, w, h} = cropRect;
+
             image
-              .autocrop()
+              .crop(x, y, w, h)
               .getBuffer(jimp.MIME_PNG, (err, buffer) => {
                 return err ? reject(err) : resolve(buffer);
-              });
+              })
+            ;
           });
         },
         (err) => Promise.reject(err)
       );
   }
 
+  getCropMatches_(image) {
+    const trimColors = [0x333333FF];
+    const rows = [];
+    const cols = [];
+
+    image.scan(0, 0, image.bitmap.width, image.bitmap.height, (x, y) => {
+      if (!rows[y]) {
+        rows[y] = [];
+      }
+      if (!cols[x]) {
+        cols[x] = [];
+      }
+      const isMatch = trimColors.includes(image.getPixelColor(x, y));
+      rows[y][x] = isMatch;
+      cols[x][y] = isMatch;
+    });
+
+    return {rows, cols};
+  }
+
+  getCropRect_({rows, cols}) {
```

Comment:
This is pretty clever. However, I wonder if something similar could be accomplished with more certainty if we had a right and bottom border that's 1px solid #abc123 or something similarly obvious and outside of the Material scope. Cropping could then rely on finding the column with that color and making sure the subsequent rows at that column have that color. If we find one that doesn't, we can check the previous row and move left from there.

Or, could use a binary search and choose the image midpoint and move left from there until we find that color. If we find it, we're on the axis with that color. Then binary search downwards from there to find the bottom right "corner" where the pixel above it is the same color, below it is not, to the left is the same color, and to the right is not.

Eh, that gets hairy in RTL because you have to reverse where you search from.

Let's stick with this for now. I can see it potentially raising problems if we introduce tests with the background color we're checking for but that's a big "if".

This is pretty dope as-is.

Reply:
Another option, though, is to provide a crop-width argument in the screenshot API. Since the fixture.scss sets the width of each column to 171px with 10px margins, you know the exact width of the content. It's a possibility to consider.

Reply:
I originally planned to use a border, but background-color sounded easier to program 😛 

Is it OK if I create a separate follow-up PR to crop based on border instead of background-color?

Reply:
100%!! This works as-is which is great. I was mostly brainstorming.